### ISSUE TITLE
Calculator prototype

### ISSUE THREAD

**Issue by lbelmore-gc** [gc-proto/covid-19-guidance#130]
Adding this ticket to track the progress on the calculator and collaborate https://github.com/gc-proto/covid-19-guidance/projects 

@ Miguel - can you add an screenshot of where it's at and highlight the major elements left to address or verify?
Tanya is working on content for this page in a separate ticket

**Comment by lbelmore-gc** [comment]
Outstanding list of items in the works


Summary table and results:

- Bold the top 3 locations for the application form (3 at random if >3 tied)

- Fix a bug in the csv download of the summary table (extra column)

- Fix the layout issues in the final results summary

Calculation refresh feature:

- Update all locations’ lockdown support summaries when the base subsidy rate is updated

Browser issues:

- On Edge: Add a manual date input

- On IE and Safari: Reinitialize the date pickers when generating new locations

**Comment by Borges-Porteous** [comment]
The browser issues have been fixed... spent most of my day on that...

The calculator now works on IE, Safari, Edge (both the older version CRA uses, and the newer versions based on chromium) and Chrome.

The lockdown support summaries are now being updated properly when the base rent subsidy rate is updated.

@Christopher-O has said he will action the 3rd bullet on the layout for the final summary for end of week.

So the outstanding items now are:
- Bolding the top 3 locations
- Fixing the .csv

**Comment by Borges-Porteous** [comment]
The csv issue and the 3 locations have both been added.

Also, although we didn't note it here, I added a 'monthly' option to the picklist for 'Interest on Commercial Interest'.

**Comment by Borges-Porteous** [comment]
Should we close this issue? Or keep it open for more updates that will probably happen as the week continues?

**Comment by lbelmore-gc** [comment]
@Borges-Porteous - let's keep it open until we get through the remaining small tweaks for this page.

Outstanding items 
- Bold the top 3 locations for the application form (3 at random if >3 tied)
- Fix a bug in the csv download of the summary table (extra column)
- Fix the layout issues in the final results summary - @Christopher-O helping with this one

New items to address:
- rent vs own - switch order
- add back in the address fields 
- Lockdown as one word for all instances
- update summary text to explain why they only get a percentage of the 25%
- issue with javascript with rounding cents -what can be done / review with LPRAB / Kevin to help determine solution

post launch:
- review making the print location summary a separate step

**Comment by Borges-Porteous** [comment]
Most of these have now been addressed. The outstanding ones include the post launch item - I know Shaun also had some recommendations around that, so it might make sense to set up a workshop of some sort?:

- @Christopher-O has made some updates to the layout of the results summary, but we still have another layout tweaks to fix for the print preview
- I did not update the lockdown support summary - this might be something to discuss with @tanyafowler. To be honest, I wasn't quite sure what had to change... the comment from Lorraine was to make sure it was clear that the number of days on lockdown was what was affecting the top up rate, but I thought that was already pretty clear from how the summary is phrased...
- issue with javascript with rounding cents - unfortunately, Kevin explained that we can't install the JS libraries (like https://github.com/MikeMcl/big.js) that we would need to use a more precise rounding, unless we switch to a non-Canada.ca/AEM platform. 
** Kevin was explaining that this was one of the main reasons he was recommending we look to a Javascript framework type solution for future interactive service delivery, because they can much more easily install libraries to handle these types of things.
** That being said, I noticed that the main issue might not have been rounding, but rather to do with the fact that the calculator uses a more precise top up rate than the summary (more decimal points), which can result in as much as 3$ extra on the maximum 75k amount per location... I am waiting for Ken and David to reply on whether or not to use the rounded or full decimal places rate.

Just to add:
- I've also resolved the top up amount recalculation issue when using the radio buttons
- The 3 locations for the application form are now appearing with full address in the step 5 summary, (instead of being bolded in the location by location summary)
- I have updated he Owned locations to require the user to select a period option before it calculates that field and there's a message in blue saying to select a period until they do so
** I'm noticing a bug when adding a location where it inherits the message from location 1, so I'll have to tweak that...
